src/libraries/System.Private.CoreLib/src/System/IO/File.cs

```diff
@@ -1007,5 +1007,45 @@ public static Task AppendAllLinesAsync(string path, IEnumerable<string> contents
                 ? Task.FromCanceled(cancellationToken)
                 : InternalWriteAllLinesAsync(AsyncStreamWriter(path, encoding, append: true), contents, cancellationToken);
         }
+
+        /// <summary>
+        /// Creates a file symbolic link identified by <paramref name="path"/> that points to <paramref name="pathToTarget"/>.
+        /// </summary>
+        /// <param name="path">The path where the symbolic link should be created.</param>
+        /// <param name="pathToTarget">The path of the target to which the symbolic link points.</param>
+        /// <returns>A <see cref="FileInfo"/> instance that wraps the newly created file symbolic link.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="path"/> or <paramref name="pathToTarget"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="path"/> or <paramref name="pathToTarget"/> is empty.
+        /// -or-
+        /// <paramref name="path"/> or <paramref name="pathToTarget"/> contains a null character.</exception>
+        /// <exception cref="IOException">A file or directory already exists in the location of <paramref name="path"/>.
+        /// -or-
+        /// An I/O error occurred.</exception>
+        public static FileSystemInfo CreateSymbolicLink(string path, string pathToTarget)
+        {
+            string fullPath = Path.GetFullPath(path);
+            FileSystem.VerifyValidPath(pathToTarget, nameof(pathToTarget));
+
+            FileSystem.CreateSymbolicLink(path, pathToTarget, isDirectory: false);
+            return new FileInfo(originalPath: path, fullPath: fullPath, isNormalized: true);
+        }
+
+        /// <summary>
+        /// Gets the target of the specified file link.
+        /// </summary>
+        /// <param name="linkPath">The path of the file link.</param>
+        /// <param name="returnFinalTarget"><see langword="true"/> to follow links to the final target; <see langword="false"/> to return the immediate next link.</param>
+        /// <returns>A <see cref="FileInfo"/> instance if <paramref name="linkPath"/> exists, independently if the target exists or not. <see langword="null"/> if <paramref name="linkPath"/> is not a link.</returns>
+        /// <exception cref="IOException">The file on <paramref name="linkPath"/> does not exist.
+        /// -or-
+        /// The link's file system entry type is inconsistent with that of its target.
+        /// -or-
+        /// Too many levels of symbolic links.</exception>
+        /// <remarks>When <paramref name="returnFinalTarget"/> is <see langword="true"/>, the maximum number of symbolic links that are followed are 40 on Unix.</remarks>
```

Comment:
since the remarks provides the value for Unix, it would be nice to do it for Windows as well

```suggestion
        /// <remarks>When <paramref name="returnFinalTarget"/> is <see langword="true"/>, the maximum number of symbolic links that are followed are 40 on Unix and 64 on Windows.</remarks>
```

Reply:
In practice this seems to be more complicated, as per the [docs](https://docs.microsoft.com/windows/win32/fileio/reparse-points) the reparse limit is 63 but on my own testing I can see that not being true. 
I got a couple conclusions from creating tests that verify the reparse point limit in windows:
* The limit is 62 when using relative paths as the target.
* The limit decreases to "half" (31) when your targets are specified as fully qualified paths i.e: absolutes.

Reply:
Regardless, I mentioned the limit of 63 in our remarks.